tests/phpunit/E2E/Extern/RestTest.php

```diff
@@ -212,6 +211,12 @@ public function testAPICalls($query, $is_error) {
     $this->updateAdminApiKey();
 
     $client = CRM_Utils_HttpClient::singleton();
+    if (CRM_Core_Config::singleton()->userFramework === 'WordPress') {
+      if (!empty($query['q'])) {
+        $query['rq'] = $query['q'];
+        unset($query['q']);
+      }
+    }
     list($status, $data) = $client->post($this->getRestUrl(), $query);
     $this->assertEquals(CRM_Utils_HttpClient::STATUS_OK, $status);
```

Comment:
There's an additional assertion that I think needs to be sprinkled in here. It's something largely taken for granted in `extern/rest.php` but becomes more important when we're plugged into request-handlers that we don't control. Specifically:

*The REST responses should **NOT** start sessions -- they should not return PHP session cookies.*

The main protocol for `extern/rest.php` is stateless. Many REST-y clients ignore cookies (because they expect protocols to be stateless). If a REST end-point is allowed to create unnecessary sessions by default, then you're liable to get a new session for *every single REST request* - which will add up to a lot of noise.